transport-msg.tex

```diff
@@ -437,3 +455,256 @@ \subsection{Messages Definition}\label{sec:Virtio Transport Options / Virtio Ove
+and the encoding of the payload for the request and answer (when applicable).
+
+\newcommand{\busdef}[1]{\paragraph{BUS_MSG_#1}\label{sec:Virtio Transport Options / Virtio Over Messages / Message Bus / Message Definition / BUS_MSG_#1}}
+
+\busdef{ERROR}
+
+The Error message is used by one side of the bus to inform the other side that
+one message could not be processed correctly. The message contains a Virtio
+Messsage error code.
+
+This message shall be used for Bus errors not specific to a particular device.
+In case of a device specific error not impacting the complete bus, the
+VIRTIO_MSG_ERROR shall be used.
+
+As for the VIRTIO_MSG_ERROR, this message supports to have a NULL terminating
+String that can be used to give more information printable to the user.
+
+This message can be received by the frontend or the backend side and can be
+send as answer to an other message or directly to signal an error.
+
+\begin{tabular}{|l|l|l|l|}
+\hline
+Type & Offset & Size (bytes) & Content \\
+\hline \hline
+Any     & 0 & 4 & Virtio Message Error Code \\
+        & 4 & 1 & Message ID which generated an error or 0 if none \\
+        & 5 & 1 & Data Type: 0 (Reserved), 1 (STRING), 2-255 (Implementation Defined) \\
+        & 6 & 32 & Null Terminated String if Data Type is STRING else Implementation Defined \\
+\hline
+\end{tabular}
+
+\busdef{GET_DEVICES}
+
+The Get Devices message is used by the frontend to retrieve the IDs of the
+devices available on the backend of the bus.
+
+The frontend requests the availability of a group of 256 device IDs and the
+backend returns a bitmap indicating which device IDs are available at the
+requested device ID offset.
+
+A request at offset X gives the availability of device IDs X*32 to (X+1)*32-1.
+
+The backend also returns a next offset indicating to the frontend what device
+ID offset should be requested next. This is to reduce the number of messages
+required to scan all IDs when the backend has device IDs far from each other.
+
+This message shall be used in the following way:
+\begin{itemize}
+\item frontend request device IDs at offset X.
+\item backend answers with the availability of device IDs X*32 to (X+1)*32-1
+  and indicates to the frontend which offset to request after using the next
+  offset.
+\end{itemize}
+The frontend starts by doing a request with offset 0 and then iterates using
+the next offset returned by the backend until the backend gives a next offset
+of 0.
+
+This message is sent by the frontend side of the bus to the backend side and
+expects an answer from the backend side.
+
+\begin{tabular}{|l|l|l|l|}
+\hline
+Type & Offset & Size (bytes) & Content \\
+\hline \hline
+Request & 0 & 2 & Offset \\
+        & 2 & 34 & Reserved (MBZ) \\
+\hline
+Answer & 0 & 2 & Offset \\
+       & 2 & 2 & Next offset or 0 if no devices after Offset \\
+       & 4 & 2 & Reserved (MBZ) \\
+       & 6 & 32 & Bit[n]: Device[offset*32 + n] not available(0)/available(1) \\
```

Comment:
the max size of the payload is 36.   except if i missed something:

```diff
Answer & 0 & 2 & Offset \\
       & 2 & 2 & Next offset or 0 if no devices after Offset \\
-       & 4 & 2 & Reserved (MBZ) \\
-       & 6 & 32 & Bit[n]: Device[offset*32 + n] not available(0)/available(1) \\
+       & 4 & 32 & Bit[n]: Device[offset*32 + n] not available(0)/available(1) \\
```

Reply:
Actually a Bus message has 38 bytes of payload, a Virtio Message only has 36 as we have the Device ID added in the header.

Reply:
We have no message requiring 38 bytes so I do not have anything against that at all.
I will do the changes for that.